src/components/common/PostalCode/PostalCodeInput.tsx

```diff
@@ -1,26 +1,62 @@
-import { useStorage } from '@faststore/sdk'
+import { useSession, useStorage } from '@faststore/sdk'
 import { Input as UIInput, Label as UILabel } from '@faststore/ui'
 import React, { useRef } from 'react'
 import type { KeyboardEvent } from 'react'
+import { gql } from '@vtex/graphql-utils'
+import { request } from 'src/sdk/graphql/request'
+import type {
+  RegionQueryQuery,
+  RegionQueryQueryVariables,
+} from '@generated/graphql'
 
 import './postal-code-input.scss'
 
 const POSTAL_CODE_STORAGE_KEY = 'main::store::postalCode'
 const POSTAL_CODE_INPUT_ID = 'postal-code-input'
 
+export const RegionQuery = gql`
+  query RegionQuery($session: IStoreSession!) {
+    session(session: $session) {
+      channel
+    }
+  }
+`
+
 export default function PostalCodeInput() {
   const ref = useRef<HTMLInputElement>(null)
   const [postalCode, setPostalCode] = useStorage<string>(
     POSTAL_CODE_STORAGE_KEY,
     ''
   )
 
-  const handleSubmit = (event: KeyboardEvent<HTMLInputElement>) => {
+  const { country, setSession, ...partialSession } = useSession()
+
+  const handleSubmit = async (event: KeyboardEvent<HTMLInputElement>) => {
     const value = ref.current?.value
 
-    if (event.key === 'Enter' && typeof value === 'string') {
-      setPostalCode(value)
+    if (!(event.key === 'Enter' && typeof value === 'string')) {
+      return
     }
+
+    setPostalCode(value)
```

Comment:
isn't it better to set the postal code after the request was sucessfull?

Reply:
Yeah, could be.